falcon/request_helpers.py

```diff
@@ -129,3 +134,271 @@ def readlines(self, hint=None):
+                the value of the param, but only if the param is present.
+
+        Returns:
+            str: The value of the param as a string, or ``None`` if param is
+                not found and is not required.
+
+        Raises:
+            HTTPBadRequest: A required param is missing from the request.
+
+        """
+
+        params = self.source
+
+        # PERF: Use if..in since it is a good all-around performer; we don't
+        #       know how likely params are to be specified by clients.
+        if name in params:
+            # NOTE(warsaw): If the key appeared multiple times, it will be
+            # stored internally as a list.  We do not define which one
+            # actually gets returned, but let's pick the last one for grins.
+            param = params[name]
+            if isinstance(param, list):
+                param = param[-1]
+
+            if store is not None:
+                store[name] = param
+
+            return param
+
+        if not required:
+            return None
+
+        raise HTTPMissingParam(name)
+
+    def get_as_int(self, name,
+                   required=False, min=None, max=None, store=None):
+        """Return the value of a query string parameter as an int.
+
+        Args:
+            name (str): Parameter name, case-sensitive (e.g., 'limit').
+            required (bool, optional): Set to ``True`` to raise
+                ``HTTPBadRequest`` instead of returning ``None`` when the
+                parameter is not found or is not an integer (default
+                ``False``).
+            min (int, optional): Set to the minimum value allowed for this
+                param. If the param is found and it is less than min, an
+                ``HTTPError`` is raised.
+            max (int, optional): Set to the maximum value allowed for this
+                param. If the param is found and its value is greater than
+                max, an ``HTTPError`` is raised.
+            store (dict, optional): A ``dict``-like object in which to place
+                the value of the param, but only if the param is found
+                (default ``None``).
+
+        Returns:
+            int: The value of the param if it is found and can be converted to
+                an integer. If the param is not found, returns ``None``, unless
+                `required` is ``True``.
+
+        Raises
+            HTTPBadRequest: The param was not found in the request, even though
+                it was required to be there. Also raised if the param's value
+                falls outside the given interval, i.e., the value must be in
+                the interval: min <= value <= max to avoid triggering an error.
+
+        """
+
+        params = self.source
+
+        # PERF: Use if..in since it is a good all-around performer; we don't
+        #       know how likely params are to be specified by clients.
+        if name in params:
```

Comment:
> This does not seem very DRY to me

I'm also a big fan of DRY. However, we may want to keep this part of the code WET since it avoids an extra function call (and possibly a few additional operations).

> I'm not sure the store kwarg pattern is really strong

It can save some typing when composing options to pass to a different layer of the app, but that aside, I'm reluctant to make a breaking change since it would break some existing apps out there.